src/hotspot/share/nmt/memLogRecorder.cpp

```diff
@@ -0,0 +1,826 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+// record pattern of allocations of memory calls:
+//
+// ./build/macosx-aarch64-server-release/xcode/build/jdk/bin/java -XX:+UnlockDiagnosticVMOptions -XX:NMTRecordMemoryAllocations=0x7FFFFFFF -jar build/macosx-aarch64-server-release/images/jdk/demo/jfc/J2Ddemo/J2Ddemo.jar
+//
+// OR record pattern of allocations of virtual memory calls:
+//
+// ./build/macosx-aarch64-server-release/xcode/build/jdk/bin/java -XX:+UnlockDiagnosticVMOptions -XX:NMTRecordVirtualMemoryAllocations=0x7FFFFFFF -jar build/macosx-aarch64-server-release/images/jdk/demo/jfc/J2Ddemo/J2Ddemo.jar
+//
+// this will produce 3 files:
+//
+// #1 hs_nmt_pid22770_allocs_record.log (is the chronological record of the the desired operations)
+// OR
+// #1 hs_nmt_pid22918_virtual_allocs_record.log (is the chronological record of the desired operations)
+// #2 hs_nmt_pid22770_info_record.log (is the record of default NMT memory overhead and the NMT state)
+// #3 hs_nmt_pid22770_threads_record.log (is the record of thread names that can be retrieved later when processing)
+//
+// then to actually run the benchmark:
+//
+// ./build/macosx-aarch64-server-release/xcode/build/jdk/bin/java -XX:+UnlockDiagnosticVMOptions -XX:NMTBenchmarkRecordedPID=43100 -XX:NMTBenchmarkRecordedLoops=10
```

Comment:
?? How does this run anything - you haven't passed an initial class to the Java invocation?

Reply:
We check for the flags when we initialize NMT and run only when NMTBenchmarkRecordedPID is set.

Reply:
We do have some terminal flags like `-version` and `-Xshare:dump`, but generally the VM does not have magic flags that behave that way. I still don't have a clear picture of the different phases involved with this work. We have a recording phase. Then we have the benchmarking phase that does something (I can't figure out what) with the recording. Then we have the Java tool which does ... what?

In any case I have concerns about how the "benchmarking" is initiated.

Reply:
I think the above comment ended up in the wrong discussion.